### ISSUE TITLE
can't download le.com

### ISSUE THREAD

**Issue by DoctorLai** [soimort/you-get#1961]
Please make sure these boxes are checked before submitting your issue – thank you!

- [ ] You can actually watch the video in your browser or mobile application, but not download them with `you-get`.
- [ ] Your `you-get` is up-to-date.
- [ ] I have read <https://github.com/soimort/you-get/wiki/FAQ> and tried to do so.
- [ ] The issue is not yet reported on <https://github.com/soimort/you-get/issues> or  <https://github.com/soimort/you-get/wiki/Known-Bugs>. If so, please add your comments under the existing issue.
- [ ] The issue (or question) is really about `you-get`, not about some other code or project.

Run the command with the `--debug` option, and paste the full output inside the fences:

```
 you-get --itag=18 -t 999 -u --no-caption --debug  http://www.le.com/ptv/vplay/2131017.html#vid=2131017
[DEBUG] get_content: http://www.le.com/ptv/vplay/2131017.html#vid=2131017
[DEBUG] get_content: http://api.letv.com/mms/out/video/playJson?id=2131017&platid=1&splatid=101&format=1&tkey=797102253&domain=www.letv.com
you-get: version 0.4.715, a tiny downloader that scrapes the web.
you-get: ['http://www.le.com/ptv/vplay/2131017.html#vid=2131017']
Traceback (most recent call last):
  File "/usr/local/bin/you-get", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.5/dist-packages/you_get/__main__.py", line 92, in main
    main(**kwargs)
  File "/usr/local/lib/python3.5/dist-packages/you_get/common.py", line 1429, in main
    script_main('you-get', any_download, any_download_playlist, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/you_get/common.py", line 1337, in script_main
    download_main(download, download_playlist, args, playlist, stream_id=stream_id, output_dir=output_dir, merge=merge, info_only=info_only, json_output=json_output, caption=caption)
  File "/usr/local/lib/python3.5/dist-packages/you_get/common.py", line 1158, in download_main
    download(url, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/you_get/common.py", line 1422, in any_download
    m.download(url, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/you_get/extractors/le.py", line 137, in letv_download
    letv_download_by_vid(vid, title=title, output_dir=output_dir, merge=merge, info_only=info_only,**kwargs)
  File "/usr/local/lib/python3.5/dist-packages/you_get/extractors/le.py", line 90, in letv_download_by_vid
    ext , urls = video_info(vid,**kwargs)
  File "/usr/local/lib/python3.5/dist-packages/you_get/extractors/le.py", line 61, in video_info
    support_stream_id = info["playurl"]["dispatch"].keys()
KeyError: 'playurl'



```

If there's anything else you would like to say (e.g. in case your issue is not about downloading a specific video; it might as well be a general discussion or proposal for a new feature), fill in the box below; otherwise, you may want to post an emoji or meme instead:

> [WRITE SOMETHING]
> [OR HAVE SOME :icecream:!]

汉语翻译最终日期：2016年02月26日

在提交前，请确保您已经检查了以下内容!

- [ ] 你可以在浏览器或移动端中观看视频，但不能使用`you-get`下载.
- [ ] 您的`you-get`为最新版.
- [ ] 我已经阅读并按 <https://github.com/soimort/you-get/wiki/FAQ> 中的指引进行了操作.
- [ ] 您的问题没有在<https://github.com/soimort/you-get/issues> , <https://github.com/soimort/you-get/wiki/FAQ> 或 <https://github.com/soimort/you-get/wiki/Known-Bugs> 报告，否则请在原有issue下报告.
- [ ] 本问题确实关于`you-get`, 而不是其他项目.

请使用`--debug`运行，并将输出粘贴在下面:

```
[在这里粘贴完整日志]
```

如果您有其他附言，例如问题只在某个视频发生，或者是一般性讨论或者提出新功能，请在下面添加;或者您可以卖个萌:

> [您的内容]
> [舔 :icecream:!]

**Comment by DoctorLai** [comment]
Thanks for so fast reply.
Yes, i noticed that and updated the comments

**Comment by DoctorLai** [comment]
Thanks,
when is this going to be published?
e.g. I use  the following command  but it hasn't shown any updates yet.

pip3 install --upgrade you-get

**Comment by skywang00** [comment]
@DoctorLai 
#1954 

> In order to get the latest develop branch without messing up the PIP, you can try:
$ pip3 install --upgrade git+https://github.com/soimort/you-get@develop

**Comment by JayXon** [comment]
I'm using latest develop branch and still have the same problem (it was working yesterday)
```
λ python C:\Users\JayXon\Documents\GitHub\you-get\you-get -i -d http://www.letv.com/ptv/vplay/29394106.html
[DEBUG] get_content: http://www.letv.com/ptv/vplay/29394106.html
[DEBUG] get_content: http://player-pc.le.com/mms/out/video/playJson?id=29394106&platid=1&splatid=101&format=1&tkey=1633572255&domain=www.le.com&region=cn&source=1000&accesyx=1
you-get: version 0.4.715, a tiny downloader that scrapes the web.
you-get: ['http://www.letv.com/ptv/vplay/29394106.html']
Traceback (most recent call last):
  File "C:\Users\JayXon\Documents\GitHub\you-get\you-get", line 11, in <module>
    you_get.main(repo_path=_filepath)
  File "C:\Users\JayXon\Documents\GitHub\you-get/src\you_get\__main__.py", line 92, in main
    main(**kwargs)
  File "C:\Users\JayXon\Documents\GitHub\you-get/src\you_get\common.py", line 1443, in main
    script_main('you-get', any_download, any_download_playlist, **kwargs)
  File "C:\Users\JayXon\Documents\GitHub\you-get/src\you_get\common.py", line 1354, in script_main
    download_main(download, download_playlist, args, playlist, output_dir=output_dir, merge=merge, info_only=info_only, json_output=json_output, caption=caption)
  File "C:\Users\JayXon\Documents\GitHub\you-get/src\you_get\common.py", line 1156, in download_main
    download(url, **kwargs)
  File "C:\Users\JayXon\Documents\GitHub\you-get/src\you_get\common.py", line 1436, in any_download
    m.download(url, **kwargs)
  File "C:\Users\JayXon\Documents\GitHub\you-get/src\you_get\extractors\le.py", line 143, in letv_download
    letv_download_by_vid(vid, title=title, output_dir=output_dir, merge=merge, info_only=info_only,**kwargs)
  File "C:\Users\JayXon\Documents\GitHub\you-get/src\you_get\extractors\le.py", line 96, in letv_download_by_vid
    ext , urls = video_info(vid,**kwargs)
  File "C:\Users\JayXon\Documents\GitHub\you-get/src\you_get\extractors\le.py", line 64, in video_info
    support_stream_id = info["playurl"]["dispatch"].keys()
KeyError: 'playurl'
```

**Comment by rosynirvana** [comment]
cannot reproduce
```
./you-get -i http://www.letv.com/ptv/vplay/29394106.html
Current Video Supports:
	--format 720p <URL>
	--format 1000 <URL>
	--format 350 <URL>
	--format 1300 <URL>
	--format 1080p <URL>
Site:       Le.com
Title:      欢乐颂2 01
Type:       MPEG-4 video (video/mp4)
Size:       1011.13 MiB (1060241792 Bytes)
```

**Comment by JayXon** [comment]
Here's the json returned:
`b'{"code":1,"msgs":{"playstatus":{"status":0,"flag":1},"statuscode":1001}}'`
Doesn't look like a time problem.

**Comment by JayXon** [comment]
找到原因了，如果Chrome里面不开unblock youku的话也会返回同样的json，应该是地域限制。
但是le.py貌似并不支持`-y proxy.uku.im:443`，用了之后完全没有效果，而如果使用`-x proxy.uku.im:443`的话，则在第一个get_content就直接403了，都到不了json那一步，有人知道为什么会403么？是不是这个代理不能用了？

**Comment by JayXon** [comment]
Somehow I couldn't get proxy.uku.im to work, so I have to use www.yōukù.com/proxy
This patch worked for me, but I don't know if you guys want to hardcode proxy urls, I can send a PR if that's ok.
```diff
diff --git a/src/you_get/extractors/le.py b/src/you_get/extractors/le.py
index bc034f3..eb87b5e 100644
--- a/src/you_get/extractors/le.py
+++ b/src/you_get/extractors/le.py
@@ -59,6 +59,11 @@ def video_info(vid,**kwargs):
     info=json.loads(str(r,"utf-8"))
     info = info['msgs']

+    if 'playurl' not in info:
+        url = 'http://www.xn--yuk-6na13b.com/proxy/http/' + url[7:]
+        r = get_content(url, decoded=False)
+        info = json.loads(str(r, "utf-8"))
+        info = info['msgs']

     stream_id = None
     support_stream_id = info["playurl"]["dispatch"].keys()

```

**Comment by soimort** [comment]
Already fixed by #1928.
